packages/sdk/src/mcp/knowledge/api.ts

```diff
@@ -293,34 +320,128 @@ export const addFileToKnowledgeBase = createKnowledgeBaseTool({
     const proccessedFile = await fileProcessor.processFile(fileUrl);
 
     const fileMetadata = {
-      ...metadata, // metadata has path that is used to get full file path
+      ..._metadata,
       ...proccessedFile.metadata,
-      fileSize: proccessedFile.metadata.fileSize.toString(),
-      chunkCount: proccessedFile.metadata.chunkCount.toString(),
+      fileSize: proccessedFile.metadata.fileSize,
+      chunkCount: proccessedFile.metadata.chunkCount,
     };
 
     const contentItems = proccessedFile.chunks.map((
       chunk: { text: string; metadata: Record<string, string> },
+      idx,
     ) => ({
       content: chunk.text,
       metadata: {
         ...fileMetadata,
         ...chunk.metadata,
+        chunkIndex: idx,
       },
     }));
 
     const docIds = await batchUpsertVectorContent(contentItems, c);
 
     assertHasWorkspace(c);
-    if (path) {
-      await c.db.from("deco_chat_assets").update({
-        metadata: {
-          ...metadata,
-          docIds,
-        },
-      }).eq("workspace", c.workspace.value).eq("file_url", path);
+
+    const metadata: z.infer<typeof KnowledgeFileMetadataSchema> = {
```

Comment:
Move some metadata to columns, file_type, knowledge_index,  file_size, chunk_count

Reply:
done